wurst/systems/meatsystem/MeatSystem.wurst

```diff
@@ -0,0 +1,379 @@
+                this.ownerUnit.issueImmediateOrderById(Orders.stop)
+            else if ownerUnit.getPos().distanceToSq(this.smartPickupTarget.getPos()) <= rangeSq
+                //Otherwise, if in range, pick up meat
+                this.tryPickupMeat(this.smartPickupTarget, true)
+                destroy this.smartPickupCallback
+                this.smartPickupCallback = null
+                this.ownerUnit.issueImmediateOrderById(Orders.stop)
+
+    function updateMeatAbilities()
+        //Swaps pick up meat / drop ability for corresponding meat amount
+        for i = 0 to 7
+            ownerUnit.removeAbility(ABILITY_IDS_DROPMEAT[i])
+
+        let meatCount = carriedMeat.size()
+        if meatCount != 0
+            if this.isTrollInventory //Avoid swapping pickup ability swap for traps, which must have both abilities at all times
+                ownerUnit.removeAbility(ABILITY_ID_PICKUPMEAT_TROLL)
+                ownerUnit.addAbility(ABILITY_ID_EAT_RAW_MEAT)
+                ownerUnit.removeAbility(ABILITY_ID_EAT_RAW_MEAT_DISABLED)
+            ownerUnit.addAbility(ABILITY_IDS_DROPMEAT[meatCount - 1])
+        else
+            if this.isTrollInventory
+                ownerUnit.removeAbility(ABILITY_ID_EAT_RAW_MEAT)
+                ownerUnit.addAbility(ABILITY_ID_EAT_RAW_MEAT_DISABLED)
+                ownerUnit.addAbility(ABILITY_ID_PICKUPMEAT_TROLL)
+
+    ondestroy
+        emptyToGround()
+        if this.smartPickupCallback != null
+            destroy this.smartPickupCallback
+        if meatInventoryInstances.has(this.ownerUnit)
+            meatInventoryInstances.remove(this.ownerUnit)
+        destroy carriedMeat
+
+function unitTryPickupMeat(unit u, item meat) returns bool
+    //Call meatpickup func for instance, or create one first and then call
+    if meatInventoryInstances.has(u)
+        return meatInventoryInstances.get(u).tryPickupMeat(meat, true)
+    else
+        let instance = new MeatInventory(u)
+        meatInventoryInstances.put(u, instance)
+        return instance.tryPickupMeat(meat, true)
+
+function unitTryPickupMeatArea(unit u)
+    //Call areapickup func for instance, or create one first and then call
+    if meatInventoryInstances.has(u)
+        meatInventoryInstances.get(u).tryPickupMeatArea()
+    else
+        let instance = new MeatInventory(u)
+        meatInventoryInstances.put(u, instance)
+        instance.tryPickupMeatArea()
+
+function onTargetOrder()
+    let order = GetIssuedOrderId()
+    let u = GetOrderedUnit()
+    if order != Orders.smart
+        return
+
+    let target = GetOrderTargetItem()
+
+    //Smart pickup system -> simulate picking up meat like an item if full inventory, like in Bannars StackNSplit system
+    if (target != null
+        and target.isPickupable()
+        and target.getTypeId() == ITEM_RAW_MEAT
+        and u.isInventoryFull()
+        and (meatInventoryInstances.has(u) or isUnitMeatCarrier(u)))
+
+        if meatInventoryInstances.has(u)
```

Comment:
Create the `instance` right here and don't put the rest of the logic in a branch.

Reply:
Changed